tool/scaffold/src/main/java/org/openscience/cdk/tools/scaffold/ScaffoldGenerator.java

```diff
@@ -0,0 +1,2412 @@
+        /*Go through all fragments created by iterative removal*/
+        for (int tmpCounter = 0; tmpCounter < tmpIterativeRemovalList.size(); tmpCounter++) {
+            IAtomContainer tmpIterMol = tmpIterativeRemovalList.get(tmpCounter); //Take the next molecule from the list
+            List<IAtomContainer> tmpAllRingsList = this.getRingsInternal(tmpIterMol, true);
+            int tmpRingSize = tmpAllRingsList.size();
+            /*Go through all rings of the fragment*/
+            for (IAtomContainer tmpRing : tmpAllRingsList) {
+                /*Skip molecule if it has less than 2 rings*/
+                if (tmpRingSize < 2) {
+                    continue;
+                }
+                /*Consider all removable terminal rings*/
+                if (this.isRingTerminal(tmpIterMol, tmpRing) && this.isRingRemovable(tmpRing, tmpAllRingsList, tmpIterMol)) {
+                    //Remove next ring
+                    IAtomContainer tmpRingRemoved = this.getScaffoldInternal(this.removeRing(tmpIterMol, true, tmpRing), true, this.determineAromaticitySetting, aromaticityModelSetting, this.scaffoldModeSetting);
+                    /*The node is not yet in the network and must therefore still be added.*/
+                    if(!tmpScaffoldNetwork.containsMolecule(tmpRingRemoved)) {
+                        tmpIterativeRemovalList.add(tmpRingRemoved);
+                        //Create new node
+                        NetworkNode<IAtomContainer> tmpNewNode = new NetworkNode<>(tmpRingRemoved);
+                        //Add the new node as parent for the old one
+                        NetworkNode tmpNode1 = (NetworkNode) tmpScaffoldNetwork.getNode(tmpIterMol);
+                        tmpNode1.addParent(tmpNewNode);
+                        //Add Origin
+                        tmpNewNode.addOriginSmiles(tmpFirstNodeSmiles);
+                        //Add new node to the tree
+                        tmpScaffoldNetwork.addNode(tmpNewNode);
+                        /*The node is already in the network*/
+                    }   else {
+                        /*Node with the same molecule already in the tree*/
+                        NetworkNode tmpOldNode = (NetworkNode) tmpScaffoldNetwork.getNode(tmpRingRemoved);
+                        /*Add parent*/
+                        NetworkNode tmpNewNode = (NetworkNode) tmpScaffoldNetwork.getNode(tmpIterMol);
+                        tmpNewNode.addParent(tmpOldNode);
+                    }
+                }
+            }
+        }
+        return tmpScaffoldNetwork;
+    }
+
+    /**
+     * Generates a network for each molecule in the list and merges the networks together. <p>
+     *
+     * Iteratively removes the terminal rings of each molecule. All resulting scaffolds of a molecule are saved in a ScaffoldNetwork.
+     * A new level is created when the total number of rings decreases by 1.
+     * Level 0 is the one with the fewest (mostly 1) rings and therefore the root. <br>
```

Comment:
What do you mean with "mostly 1" here?

Reply:
I have added more details on the latest commit.